openwpm/browser_manager.py

```diff
@@ -394,33 +401,32 @@ def shutdown_browser(self, during_init: bool, force: bool = False) -> None:
         self.close_browser_manager(force=force)
 
         # Archive browser profile (if requested)
-        if not during_init and self.browser_params.profile_archive_dir is not None:
-            self.logger.warning(
-                "BROWSER %i: Archiving the browser profile directory is "
-                "currently unsupported. "
-                "See: https://github.com/mozilla/OpenWPM/projects/2" % self.browser_id
-            )
-        """
         self.logger.debug(
-            "BROWSER %i: during_init=%s | profile_archive_dir=%s" % (
-                self.browser_id, str(during_init),
-                self.browser_params.profile_archive_dir)
+            "BROWSER %i: during_init=%s | profile_archive_dir=%s"
+            % (
+                self.browser_id,
+                str(during_init),
+                self.browser_params.profile_archive_dir,
+            )
         )
-        if (not during_init and
-                self.browser_params.profile_archive_dir is not None):
+        if not during_init and self.browser_params.profile_archive_dir is not None:
             self.logger.debug(
-                "BROWSER %i: Archiving browser profile directory to %s" % (
-                    self.browser_id,
-                    self.browser_params.profile_archive_dir))
-            profile_commands.dump_profile(
-                self.current_profile_path,
-                self.manager_params,
-                self.browser_params,
-                self.browser_params.profile_archive_dir,
+                "BROWSER %i: Archiving browser profile directory to %s"
+                % (self.browser_id, self.browser_params.profile_archive_dir)
+            )
+            tar_path = self.browser_params.profile_archive_dir / "profile.tar.gz"
+            self.browser_params.profile_path = self.current_profile_path
+            dump_profile_command = DumpProfileCommand(
+                tar_path=tar_path,
                 close_webdriver=False,
-                compress=True
+                compress=True,
+            )
+            dump_profile_command.execute(
+                webdriver=None,
+                browser_params=self.browser_params,
+                manager_params=self.manager_params,
+                extension_socket=None,
             )
```

Comment:
As I understand this, we can't use `DumpProfileCommand` as part of a command sequence here and in `launch_browser_manager`. Our only choice is to call its `execute()` method directly. Passing `None` to  `webdriver` and `extension_socket` looks a bit weird though. I also had to change the type of `extension_socket` from `ClientSocket` to `Optional[ClientSocket]` in `DumpProfileCommand`'s `execute()` method for mypy to stop complaining.
Do you have a suggestion on how to do this in a better way?

Reply:
Hmm, not really.
Maybe we can extract the actual implementation of the DumpProfileCommand.execute method into a free function and then call it from here and the Command.

Reply:
> Maybe we can extract the actual implementation of the DumpProfileCommand.execute method into a free function and then call it from here and the Command.

This sounds like the right solution to me.